src/test/java/com/openlattice/rehearsal/entitysets/EntitySetsTest.kt

```diff
@@ -63,4 +73,133 @@ class EntitySetsTest : MultipleAuthenticatedUsersBase() {
                 "Linked entity sets is empty"
         )
     }
+
+    @Test
+    fun testCreateBadDataExpiration() {
+        //set expiration policy with negative duration of time until data expires
+        val badTTL = -10L
+        assertException({ DataExpiration(badTTL, ExpirationBase.LAST_WRITE, DeleteType.Hard) },
+                "Time until data expiration must not be negative")
+
+        //set expiration policy without required start date
+        val tTL = 10L
+        assertException({ DataExpiration(tTL, ExpirationBase.DATE_PROPERTY, DeleteType.Soft) },
+                "Must provide property type for expiration calculation")
+    }
+
+    @Test
+    fun testAddExpirationPolicy() {
+        val es = createEntitySet()
+        Assert.assertNull(es.expiration) // default entity set has no expiration policy
+
+        //set expiration policy
+        val tTL = 10L
+        val expirationPolicy = DataExpiration(tTL, ExpirationBase.FIRST_WRITE, DeleteType.Hard)
+        val update = MetadataUpdate(Optional.empty(), Optional.empty(), Optional.empty(),
+                Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(),
+                Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(),
+                Optional.empty(), Optional.of(expirationPolicy))
+        entitySetsApi.updateEntitySetMetadata(es.id, update)
+        val es2 = entitySetsApi.getEntitySet(es.id)
+        Assert.assertEquals(tTL, es2.expiration.timeToExpiration)
+        Assert.assertEquals(ExpirationBase.FIRST_WRITE, es2.expiration.expirationBase)
+        Assert.assertTrue(es2.expiration.startDateProperty.isEmpty)
+    }
+
+    @Test
+    fun testRemoveExpirationPolicy() {
+        val es = createEntitySet()
+
+        //set an expiration policy
+        val tTL = 10L
+        val expirationPolicy = DataExpiration(tTL, ExpirationBase.LAST_WRITE, DeleteType.Hard)
+        val update = MetadataUpdate(Optional.empty(), Optional.empty(), Optional.empty(),
+                Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(),
+                Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(),
+                Optional.empty(), Optional.of(expirationPolicy))
+        entitySetsApi.updateEntitySetMetadata(es.id, update)
+        val es2 = entitySetsApi.getEntitySet(es.id)
+
+        //remove expiration policy
+        entitySetsApi.removeDataExpirationPolicy(es2.id)
+        val es3 = entitySetsApi.getEntitySet(es.id)
+        Assert.assertNull(es3.expiration)
+    }
+
+    @Test
+    fun testGetExpiringEntitiesByFirstWrite() {
+        //create data to expire
+        val es = createEntitySet(personEt)
+        val entries = (1..10)
+                .map { mapOf(EdmTestConstants.personGivenNameId to setOf(RandomStringUtils.randomAscii(5))) }
+        dataApi.createEntities(es.id, entries)
+
+        //set first_write expiration policy
+        val tTL: Long = 24 * 60 * 60 * 1000 //one day
```

Comment:
Same type of thing with this long -- you can save it as a constant up top to avoid doing this math multiple times